packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -83,7 +83,7 @@ class Observatory {
     if (isolates.isNotEmpty)
       return isolates.first;
 
-    _waitFirstIsolateCompleter = new Completer<IsolateRef>();
+    _waitFirstIsolateCompleter ??= new Completer<IsolateRef>();
```

Comment:
I may change the implementation around this to something like a notifying list. Clients essentially need an isolate to talk to. If there's one already, that one works. Otherwise clients need to wait until one is available. Using a final completer would return the first matching isolate, even after that isolate had terminated.